docs/6-concepts.md

```diff
@@ -0,0 +1,192 @@
+
+
+#### POST [/api/administration/connectors/bpnDiscovery]
+
+Create new numbers (e.g., OEN, batteryID, etc.) corresponding to BPN
+endpoints as single entry.
+
+**Request body**
+
+```json
+{
+  "type": "oen",
+  "key": "oen-123"
+}
+```
+**Response**
+
+```json
+{
+  "type": "oen",
+  "key": "oen-1243",
+  "value": "bpn-123",
+  "resourceId": "1ca6f9b5-8e1d-422a-8541-9bb2cf5fe485"
+}
+```
+This service is allowed to **administrators only**.
+
+
+#### POST [/api/administration/connectors/bpnDiscovery/batch]
+
+Create new numbers (e.g., OEN, batteryID, etc.) corresponding to BPN
+endpoints as batch.
+
+**Request body**
+
+```json 
+[
+  {
+    "type": "oenId", 
+    "key": "oenId-123"
+  }, 
+  {
+    "type": "oen", 
+    "key": "oen-4444"
+  }, 
+  {
+    "type": "bpid", 
+    "key": "bpid-123"
+  }
+]
+```
+
+**Response**
+
+```json
+[
+  {
+    "message": "Validation failed.",
+    "details": {
+      "type": "type oenId is not allowed."
+    }, 
+    "key": "oenId-123", 
+    "status": 400
+  }, 
+  {
+    "message": "BpnDiscovery successfully created", 
+    "details": null, 
+    "key": "oen-4444", 
+    "status": 200
+  }, 
+  {
+    "message": "BpnDiscovery successfully created", 
+    "details": null, 
+    "key": "bpid-123", 
+    "status": 200
+  }
+]
+```
+
+This service is allowed to **administrators only**.
+
+
+#### POST [/api/administration/connectors/bpnDiscovery/search]
+
+Find BPN endpoints corresponding to numbers (e.g., OEN, batteryID, etc.)
+
+**Request body**
+
+```json
+{"searchFilter": 
+ [
+  {
+    "type": "oen", 
+    "keys": ["oen-1243",
+      "oen-11"]
+  }, 
+  {
+    "type": "bpid", 
+    "keys": [
+      "bpid-1243", 
+      "bpid-11"]
+  }
+ ]
+}
+```
+
+**Response**
+
+```json
+{
+ "bpns": 
+ [
+   {
+     "type": "oen", 
+     "key": "oen-1243", 
+     "value": "bpn-123", 
+     "resourceId": "1ca6f9b5-8e1d-422a-8541-9bb2cf5fe485"
+   },
+   {
+     "type": "oen", 
+     "key": "oen-11", 
+     "value": "bpn-123",
+     "resourceId": "972262d7-7e05-4578-936f-de236d7feb94"
+   }, 
+   {
+  "type": "bpid", 
+     "key": "bpid-1243", 
+     "value": "bpn-444", 
+     "resourceId": "1b754aeb-c753-4adf-ae6d-52842f5a38b7"
+   }, 
+   {
+     "type": "bpid", 
+     "key": "oen-11", 
+     "value": "bpn-444", 
+     "resourceId": "e43f9cf8-f0d4-4c51-b7fe-b68ea4aeea7c"
+   }
+ ]
+}
+```
+
+#### DELETE [/api/administration/connectors/bpnDiscovery/{resourceId}]
+
+Delete numbers (e.g., OEN, batteryID, etc.) corresponding to BPN
+endpoints as single entry.
+
+This service is allowed to **administrators only**.
+
+Path variable {resourceId} is required
+
+Response: http Status Code 204 -> no response
+
+An OpenApi.yaml for this service can be found under
+"backend/src/main/resources/static/bpn-discovery-service-openapi.yaml"
+
+### Security / Safety / Use the JWT Token to get the BPN from requester
+
+The whole environment is secured with Keycloak. The BPN Discovery has a
```

Comment:
Let's rather write OAuth2 here, and that we recommend using Keycloak. Keycloak just implements the OAuth2 standard.